src/nginx.py

```diff
@@ -26,6 +77,11 @@ def __init__(self, charm: ops.CharmBase, config):
         self.package = apt.add_package(PACKAGE, update_cache=True)
         self.charm = charm
         self.config = config
+        self.contexts: Dict[str, ConfigurationContext] = {
+            "main": ConfigurationContext(MAIN_CONTEXT_DEFAULTS, {}),
+            "events": ConfigurationContext(EVENTS_CONTEXT_DEFAULTS, {}),
+            "http": ConfigurationContext(HTTP_CONTEXT_DEFAULTS, {}),
+        }
```

Comment:
i don't really see why this has to be an attr of the NginxConfigurer.  Looks like it's only used by the `configure_daemon` method.  Let's move it into there

```suggestion
```